packages/tempus/src/Tempus_TimeEventBase.hpp

```diff
@@ -38,79 +38,189 @@ class TimeEventBase
+     *
+     *  For TimeEventBase, always return the default index.
+     *
+     *  \param index [in] Input index.
+     *  \return Index of the next event.
+     */
     virtual int indexOfNextEvent(int index) const
     { return defaultIndex_; }
 
-    /// Test if an event occurs within the time range.
+    /** \brief Test if an event occurs within the index range.
+     *
+     *  Find if an event is within the input range, inclusively
+     *  ( index1 <= event <= index2 ).  This may require testing
+     *  each event in the TimeEvent.
+     *
+     *  For TimeEventBase, always return false since it has no events.
+     *
+     *  \param index1 [in] Input index of one end of the range.
+     *  \param index2 [in] Input index of the other end of the range.
+     *
+     *  \return True if a index event is within the range.
+     */
     virtual bool eventInRangeIndex(int index1, int index2) const
     { return false; }
 
     /// Describe member data.
-    virtual void describe() const
+    virtual void describe(Teuchos::FancyOStream          &out,
+                          const Teuchos::EVerbosityLevel verbLevel) const
     {
-      Teuchos::RCP<Teuchos::FancyOStream> out =
-        Teuchos::VerboseObjectBase::getDefaultOStream();
-      out->setOutputToRootOnly(0);
-      *out << "TimeEventBase name = " << getName() << std::endl;
+      auto l_out = Teuchos::fancyOStream( out.getOStream() );
+      Teuchos::OSTab ostab(*l_out, 2, "TimeEventBase");
+      l_out->setOutputToRootOnly(0);
+
+      *l_out << "TimeEventBase name = " << getName() << std::endl;
     }
   //@}
 
   /// \name Accessor methods
   //@{
+    /** \brief Return the name of the TimeEvent.
+     *
+     *  The name of the TimeEvent can be used to identify
+     *  specific TimeEvents in order to take action related
+     *  to that TimeEvent (e.g., a name of "Output Special"
+     *  may indicate to output some special information).
+     *
+     *  \return The name of the TimeEvent.
+     */
     virtual std::string getName() const { return name_; }
+    /// Set the name of the TimeEvent.
     virtual void setName(std::string name) { name_ = name; }
-
+    /** \brief Return the type of TimeEvent.
+     *
+     *  Each derived class of TimeEventBase has a type that
+     *  can be used to identify the type of the TimeEvent.
+     */
+    virtual std::string getType() const { return timeEventType_; }
+    /// Return the default time used for TimeEvents.
     virtual Scalar getDefaultTime () const { return defaultTime_; }
+    /// Return the default tolerance used by TimeEvents.
     virtual Scalar getDefaultTol  () const { return defaultTol_; }
+    /// Return the default index used by TimeEvents.
     virtual int    getDefaultIndex() const { return defaultIndex_; }
   //@}
 
 
+  /// Return ParameterList with current values.
+  virtual Teuchos::RCP<const Teuchos::ParameterList> getValidParameters() const
+  { return Teuchos::null; }
```

Comment:
If you still want to construct this with a PL, why not have a valid PL? It help to know what is valid and helps with documentation too

Reply:
The TimeEventBase is really a no-op base class, so a PL really does not have any information or events.  That is why I did not worry about creating a valid PL.  But it is possible, so I went ahead and filled out getValidParameters().